### PR TITLE
Parser Improvement to ^

### PR DESCRIPTION
**Pull request by ShikharJ**

### CODE REVIEW

**Comment by isuruf** [comment]
Thanks for doing this. Is it possible to have an option `convert_xor` like in sympy?

**Comment by ShikharJ** [comment]
I'll have to add an option to parse. Shouldn't be much. I'll update if you concur?

**Comment by isuruf** [comment]
If you could that would be great.

File: symengine/parser.cpp [hunk 1]
```diff
@@ -517,7 +528,7 @@ class ExpressionParser
 
 public:
     // does all the preprocessing related to parsing
-    RCP<const Basic> parse_expr(const std::string &in)
+    RCP<const Basic> parse_expr(const std::string &in, bool convert_xor)
```
Comment (isuruf) [inline symengine/parser.cpp]: You can store this so that you don't have to pass it around.

File: symengine/parser.cpp [hunk 2]
```diff
@@ -626,9 +636,13 @@ class ExpressionParser
...
 };
 
-RCP<const Basic> parse(const std::string &s)
+RCP<const Basic> parse(const std::string &s, bool convert_xor)
 {
     ExpressionParser p;
+    if (convert_xor) {
+        p.op_precedence["^"] = p.op_precedence["**"];
+    }
+    p.convert_xor = convert_xor;
```
Comment (isuruf) [inline symengine/parser.cpp]: It's better if the above 4 lines are moved to the constructor of `ExpressionParser`

File: symengine/cwrapper.cpp [hunk 3]
```diff
@@ -485,10 +485,14 @@ CWRAPPER_OUTPUT_TYPE basic_assign(basic a, const basic b)
     CWRAPPER_END
 }
 
-CWRAPPER_OUTPUT_TYPE basic_parse(basic b, const char *str)
+CWRAPPER_OUTPUT_TYPE basic_parse(basic b, const char *str, int convert_xor)
```
Comment (isuruf) [inline symengine/cwrapper.cpp]: To keep backward compatibility, this can be a new function `basic_parse2`

File: symengine/cwrapper.cpp [hunk 4]
```diff
@@ -484,6 +484,13 @@ CWRAPPER_OUTPUT_TYPE basic_parse(basic b, const char *str)
     CWRAPPER_END
 }
 
+CWRAPPER_OUTPUT_TYPE basic_parse2(basic b, const char *str)
```
Comment (isuruf) [inline symengine/cwrapper.cpp]: I would add `int convert_xor` as a parameter here too.

**Pull request merged**